website/docs/bestpractices/analytics/trino-best-practices.md

```diff
@@ -0,0 +1,642 @@
+
+### VPC CNI Addon
+The Amazon VPC CNI plugin can be customized to manage IP address allocation efficiently. By default, the Amazon VPC CNI assigns two Elastic Network Interfaces (ENIs) per node. These ENIs reserve numerous IP addresses, particularly on larger instance types. Since Trino typically needs only one pod per node plus a few IP addresses for DaemonSet pods (for logging and networking), you can configure the CNI to limit IP address allocation and reduce overhead.
+
+The following settings to the VPC CNI Addon can be applied using the EKS API, Terraform, or any other Infrastructure-as-Code (IaC) tool. For more in-depth details, refer to the official documentation:VPC CNI Prefix and IP Target.
+
+### Configuring the VPC CNI Addon
+
+Limit IP Addresses per Node by adjusting the configuration to allocate only the required number of IPs:
+- MINIMUM_IP_TARGET: Set this to the expected number of pods per node (e.g., 30).
+- WARM_IP_TARGET: Set to 1 to keep the warm IP pool minimal.
+- ENABLE_PREFIX_DELEGATION: Improve IP address efficiency by assigning IP prefixes to worker nodes rather than individual secondary IP addresses. This approach reduces Network Address Usage (NAU) within your VPC by utilizing a smaller, more concentrated pool of IP addresses.
+
+#### Sample VPC CNI Configuration
+
+```
+vpc-cni = {
+  preserve = true
+  configuration_values = jsonencode({
+    env = {
+      MINIMUM_IP_TARGET           = "30"
+      WARM_IP_TARGET              = "1"
+      ENABLE_PREFIX_DELEGATION    = "true"
+    }
+  })
+}
+```
+
+:::tip[Key Recommendations]
+:::
+<b>Enable prefix delegation:</b> VPC CNI plugin supports prefix delegation, which assigns blocks of 16 IPv4 addresses (/28 prefix) to each node. This feature reduces the number of ENIs required per node. By using prefix delegation, you can lower your EC2 Network Address Usage (NAU), reduce network management complexity, and decrease operational costs.
+
+#### For more best practices related to Networking, [refer to our Networking guide](../networking/networking.md)
+
+</CollapsibleContent>
+
+<CollapsibleContent header={<span><h2>Storage Best Practices</h2></span>}>
+
+This sections focuses on AWS services for optimal storage management with Trino on EKS.
+
+## Amazon S3 as Primary Storage
+
+- Use S3 Standard for frequently accessed data in Trino queries
+- Implement S3 Intelligent-Tiering for data with varying access patterns
+- Enable S3 server-side encryption (SSE-S3 or SSE-KMS) for data at rest
+- Configure appropriate bucket policies and access through IAM roles
+- Use S3 bucket prefixes strategically for better query performance
+
+### EBS Storage for Coordinator and Workers
+
+- Use gp3 EBS volumes for better performance/cost ratio
+- Enable EBS encryption using KMS
+- Size EBS volumes based on spill directory requirements
+- Consider using EBS snapshots for backup strategy
+
+### Performance Optimization
```

Comment:
Now that Trino support is enabled for S3 Tables and Iceberg REST Catalog, can we list it as an option in addition to General S3 buckets?

Reply:
Added